### PR TITLE
Notifies ghosts when bombs and C4/X4 get activated

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
tweak: Ghosts now get notified whenever a bomb or C4/X4 get activated.
/:cl:

This notifies ghosts whenever a bomb gets activated and lets them jump to it. It's always hilarious watching people try to defuse bombs, or Ricky playing out hostage situations.

Edit: I added a notification for C4/X4 as well.

### CODE REVIEW

**Comment by tuypo1** [comment]
Best thing is when somebody accidentally attaches a bomb to themselves.

**Comment by 81Denton** [comment]
Actually, adding that for C4 and X4 might be nice too. It's not like every round has a ton of them that would spam ghost chat.

**Comment by tuypo1** [comment]
The question is would grenades be too much.
Maybe have syndicate grenades notify but not flashbangs and things.

**Comment by 81Denton** [comment]
I think that grenades might be too much since they can be spammed hard (like the nuke ops grenade belt) and usually just have a 5 second fuse.

**Comment by Vidiveni** [comment]
Could do the same for bomb cores. They can be built empty though.

File: code/game/machinery/syndicatebomb.dm [hunk 1]
```diff
@@ -191,6 +191,7 @@
 	next_beep = world.time + 10
 	detonation_timer = world.time + (timer_set * 10)
 	playsound(loc, 'sound/machines/click.ogg', 30, 1)
+	notify_ghosts("A [name] has been activated at [get_area(src)]!", source = src, action = NOTIFY_ORBIT, ghost_sound = 'sound/machines/beep.ogg')
```
Comment (SpaceManiac) [inline code/game/machinery/syndicatebomb.dm]:
```suggestion
	notify_ghosts("\A [src] has been activated at [get_area(src)]!", source = src, action = NOTIFY_ORBIT, ghost_sound = 'sound/machines/beep.ogg')
```

File: code/game/objects/items/grenades/plastic.dm [hunk 2]
```diff
@@ -114,7 +114,9 @@
 		target = AM
 
 		message_admins("[ADMIN_LOOKUPFLW(user)] planted [name] on [target.name] at [ADMIN_VERBOSEJMP(target)] with [det_time] second fuse")
-		log_game("[key_name(user)] planted [name] on [target.name] at [AREACOORD(user)] with [det_time] second fuse")
+		log_game("[key_name(user)] planted [name] on [target.name] at [AREACOORD(user)] with a [det_time] second fuse")
+
+		notify_ghosts("[user] has planted [name] on [target.name] with a [det_time] second fuse!", source = src, action = NOTIFY_ORBIT, ghost_sound = 'sound/machines/beep.ogg')
```
Comment (SpaceManiac) [inline code/game/objects/items/grenades/plastic.dm]:
```suggestion
		notify_ghosts("[user] has planted \a [src] on [target] with a [det_time] second fuse!", source = src, action = NOTIFY_ORBIT, ghost_sound = 'sound/machines/beep.ogg')
```

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
Avoid using "[name]" in user-facing strings

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
The orbit links don't work because they're set to `source = src`, which is summarily deleted

**Comment by 81Denton** [comment]
I changed the source to `target` now.

**Review by SpaceManiac — APPROVED** [review]

**Comment by Hathkar** [comment]
Any chance you could lower the volume on this? It's extremely loud, and gets grating very quickly.

**Comment by 81Denton** [comment]
Sure thing! I'll get to it when I'm back home.

**Pull request merged**